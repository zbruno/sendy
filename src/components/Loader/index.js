import React from 'react';
import PropTypes from 'prop-types';

const Loader = ({ color, size, utilityClasses }) => (
  <div className={`inline-block h-${size} w-${size} ${utilityClasses}`}>
    <div className={`absolute opacity-100 ${color} border-2 rounded-full animation-ripple-4 animation-1s animation-cubic-bezier animation-infinite`} />
    <div className={`absolute opacity-100 ${color} border-2 rounded-full animation-ripple-4 animation-1s animation-cubic-bezier animation-infinite animation-delay-.5s`} />
  </div>
);

Loader.propTypes = {
  size: PropTypes.number,
  utilityClasses: PropTypes.string,
  color: PropTypes.oneOf([
    'border-purple-500',
    'border-white',
    'border-gray-500',
  ]),
};

Loader.defaultProps = {
  size: 20,
  color: 'border-purple-500',
  utilityClasses: '',
};

export default Loader;
