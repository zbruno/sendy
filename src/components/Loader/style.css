.inner-loader {
  @apply absolute opacity-100 border-2 rounded-full animation-ripple-4 animation-1s animation-cubic-bezier animation-infinite;
}