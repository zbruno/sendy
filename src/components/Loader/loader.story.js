import React from 'react';

import Loader from '.';

export default {
  title: 'Loader',
  component: Loader,
  parameters: {
    jest: [],
  },
};

export const loaders = () => (
  <div className="relative flex flex-col items-start">
    <Loader size={6} color="border-purple-500" />
  </div>
);
