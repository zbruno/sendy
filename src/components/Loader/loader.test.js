import React from 'react';
import renderer from 'react-test-renderer';

import Loader from './index';

describe('Loader UI snapshots', () => {
  test('renders purple', () => {
    const component = renderer.create(
      <Loader
        size={6}
        color="border-purple-500"
      />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
