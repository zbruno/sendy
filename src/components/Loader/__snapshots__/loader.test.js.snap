// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader UI snapshots renders purple 1`] = `
<div
  className="inline-block h-6 w-6 "
>
  <div
    className="inner-loader border-purple-500"
  />
  <div
    className="inner-loader border-purple-500 animation-delay-.5s"
  />
</div>
`;
